Topcoder/TCPhoneHome.cc

```diff
@@ -0,0 +1,45 @@
+class TCPhoneHome {
+	public:
+	long long validNumbers(int digits, vector <string> specialPrefixes) {
+		int n = specialPrefixes.size();
+		map <string, bool> mp;
```

Comment:
A **set** is usually better when you want to check for inclusion (more efficient), also the intent of the code is more clear.
Please rename the variable **mp** to something more meaningful.